src/libsyntax/ast_map/mod.rs

```diff
@@ -86,7 +86,31 @@ impl<'a> Iterator for LinkedPath<'a> {
 }
 
 /// The type of the iterator used by with_path.
-pub type PathElems<'a, 'b> = iter::Chain<iter::Cloned<slice::Iter<'a, PathElem>>, LinkedPath<'b>>;
+#[derive(Clone)]
+pub struct PathElems<'a, 'b> {
```

Comment:
Oh dear the instability of `Cloned` is actually a mistake and it's intended to be `#[stable]` (in which case this change may not be required)